src/Microsoft.TestPlatform.Utilities/CodeCoverageDataAttachmentsHandler.cs

```diff
@@ -56,20 +78,16 @@ public async Task<ICollection<AttachmentSet>> ProcessAttachmentSetsAsync(ICollec
 
         private async Task<string> MergeCodeCoverageFilesAsync(IList<string> files, IProgress<int> progressReporter, CancellationToken cancellationToken)
         {
-            if (files.Count == 1)
```

Comment:
By removing this we assume the merging lib properly handles merging only one file. I'm sure it does, but should we keep this protection just in case ?

We also avoid any additional overhead, however small, from calling the lib method.

Reply:
I've added logic to function which is calling this method to invoke it only if at least 2 reports